pkg/topology/kademlia/kademlia.go

```diff
@@ -1260,7 +1260,7 @@ func nClosePeerInSlice(peers []swarm.Address, addr swarm.Address, spf sanctioned
 }
 
 func (k *Kad) IsReachable() bool {
-	return k.reachability == p2p.ReachabilityStatusPublic
+	return k.reachability != p2p.ReachabilityStatusPrivate
```

Comment:
it would be nice to know why this logic was changed, event better to have this documented here. 
someone reading this logic may assume that there is error in logic, because node will be reachable when `reachability` is `p2p.ReachabilityStatusUnknown`.

Reply:
So this is a temporary fix as I mentioned in the description. The reason is, in the private network, nodes only connect to other nodes in the private network. As a result, they are never "public". This is sort of a hack for this branch. I will revert it before checkin.

Reply:
okay got it

how do you keep track of things that will need to be changed before/after this branch is merged to master?